packages/jellyfish-api-core/__tests__/category/loan/tokenSplits.test.ts

```diff
@@ -0,0 +1,966 @@
+  const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+  expect(attributes.ATTRIBUTES[`v0/poolpairs/${ppTokenIDBefore}/token_a_fee_pct`]).not.toBeDefined()
+  expect(attributes.ATTRIBUTES[`v0/poolpairs/${ppTokenIDBefore}/token_b_fee_pct`]).not.toBeDefined()
+
+  const ppTokenIDNew = Object.keys(await testing.rpc.token.getToken(ppInfoBefore.symbol))[0]
+  const splitTokenIDNew = Object.keys(await testing.rpc.token.getToken(splitTokenSymbol))[0]
+
+  expect(attributes.ATTRIBUTES[`v0/poolpairs/${ppTokenIDNew}/token_a_fee_pct`]).toBeDefined()
+  expect(attributes.ATTRIBUTES[`v0/poolpairs/${ppTokenIDNew}/token_b_fee_pct`]).toBeDefined()
+
+  const poolpairNew = await testing.poolpair.get(ppTokenIDNew)
+  expect(poolpairNew.symbol).toStrictEqual(ppInfoBefore.symbol)
+  expect(poolpairNew.status).toStrictEqual(true)
+  expect(poolpairNew.idTokenA === ppInfoBefore.idTokenA || poolpairNew.idTokenA === splitTokenIDNew).toBeTruthy()
+  expect(poolpairNew.idTokenB === ppInfoBefore.idTokenB || poolpairNew.idTokenB === splitTokenIDNew).toBeTruthy()
+
+  if (!poolpairNew.reserveA.eq(ppInfoBefore.reserveA)) {
+    expect(poolpairNew.reserveA).toStrictEqual(ppInfoBefore.reserveA.multipliedBy(2))
+  }
+  if (!poolpairNew.reserveB.eq(ppInfoBefore.reserveB)) {
+    expect(poolpairNew.reserveB).toStrictEqual(ppInfoBefore.reserveB.multipliedBy(2))
+  }
+
+  expect(poolpairNew.tradeEnabled).toStrictEqual(true)
+}
+
+async function setupPools (): Promise<void> {
+  // create TSLA-DFI
+  await testing.poolpair.create({
+    tokenA: 'TSLA',
+    tokenB: 'DFI'
+  })
+  await testing.generate(1)
+
+  // add TSLA-DFI
+  await testing.poolpair.add({
+    a: { symbol: 'TSLA', amount: 100 },
+    b: { symbol: 'DFI', amount: 200 }
+  })
+  await testing.generate(1)
+
+  const ppTokenID = Object.keys(await testing.rpc.token.getToken('TSLA-DFI'))[0]
+
+  // setup govvars
+  await testing.rpc.masternode.setGov({
+    ATTRIBUTES: {
+      [`v0/poolpairs/${ppTokenID}/token_a_fee_pct`]: '0.01',
+      [`v0/poolpairs/${ppTokenID}/token_b_fee_pct`]: '0.03'
+    }
+  })
+
+  await testing.rpc.masternode.setGov({ LP_SPLITS: { [Number(ppTokenID)]: 1 } })
+  await testing.rpc.masternode.setGov({ LP_LOAN_TOKEN_SPLITS: { [Number(ppTokenID)]: 1 } })
+  await testing.generate(1)
+}
+
+describe('Token splits', () => {
+  beforeEach(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    await setup()
+  })
+
+  afterEach(async () => {
+    await testing.container.stop()
+  })
+
+  it('should check token after split', async () => {
+    const tokenTSLA = await testing.rpc.token.getToken(tslaID)
+    const newMintedTSLA = tokenTSLA[tslaID].minted.multipliedBy(2)
+
+    // set a token split at current block + 2 for TSLA and FB
+    const splitBlock = await testing.rpc.blockchain.getBlockCount() + 2
+    await testing.rpc.masternode.setGov({
+      ATTRIBUTES: {
+        [`v0/oracles/splits/${splitBlock}`]: `${tslaID}/2`
```

Comment:
wondering we can split the token 10 times?? :thinking:

Reply:
yes, the multiplier can be any integer theoretically.